doc/nrf/protocols/thread/device_types.rst

```diff
@@ -0,0 +1,133 @@
+.. _thread_device_types:
+
+Thread device types
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Thread devices can be either Full Thread Devices (FTD) or Minimal Thread Devices (MTD).
+The devices are divided based on their ability to exist in a network without a parent, determined by whether the device maintains a routing table (doesn't need a parent) or not (needs a parent).
+FTDs are the devices that maintain a routing table, while MTDs do not, forwarding all their messages to a parent instead.
+
+Another type of Thread device classification is based on the device's ability to route messages:
+
+* Routers can route messages, and they can function as parents to other Thread devices.
+* End Devices cannot route messages, and they cannot function as parents.
+
+Based on this classification, FTDs can be either Routers or End Devices, while MTDs can only be End Devices.
+
+For additional information about Thread device types, see the documentation for `Device Types on OpenThread portal`_.
+
+.. _thread_types_ftd:
+
+Full Thread Devices
+*******************
+
+In order to maintain the routing table, FTDs keep their radio on at all times.
+An important difference between FTDs and MTDs is that FTDs subscribe to special "all-routers" multicast addresses (``ff02::2`` and ``ff03::2``, see `IPv6 multicast addressing`_).
+Because of these reasons, they typically consume more power than MTDs.
+
+Full Thread Devices can be further divided into three categories:
+
+* Router
+* Full End Device (FED)
+* Router Eligible End Device (REED)
+
+.. list-table:: Full Thread Device categories
+   :widths: 15 10 10 15
+   :header-rows: 1
+
+   * -
+     - Router
+     - Full End Device
+     - Router Eligible End Device
+   * - Maintains a routing table
+     - Yes
+     - Yes
+     - Yes
+   * - Routes messages
+     - Yes
+     - No
+     - When working as a Router
+   * - Can extend the network
+     - Yes
+     - No
+     - When working as a Router
+
+Routers can be regarded as the backbone of a Thread network.
+They maintain the routing table and forward messages to other devices.
+Routers, unlike End Devices, can also be used to extend the network range and a Router is required for a Minimal End Device to join the network.
+The maximum number of Routers in a single Thread network is 32.
+
+FEDs maintain a routing table, but they can't route messages.
+This means they don't require a parent (Router) to function, but they cannot become parents themselves.
+
+REEDs maintain a routing table and, unlike FEDs, they can become Routers if needed.
+If a REED is the only device in range of an End Device trying to join the network, it will promote itself to a Router (see `Joining an existing Thread network`_ in the OpenThread documentation for more information).
+Conversely, when a Router has no children it can downgrade itself to a REED.
```

Comment:
It's true, but also simplification. The Thread specification provides a few other possibilities when Routers should downgrade to REED role. Basically it's all about the network topology, so maybe:

"Depending on the network topology conditions, Router may downgrade itself to a REED role (e.g. when does not have any children and sufficient number of routers are already in the network)."